lib/net/x86-64-linux/Net.v3

```diff
@@ -327,6 +537,45 @@ component Net {
 		return (if(ret.0 >= 0, int.view(ret.0), -1), ptr);
 	}
 
+	// Send a message {msg} to the host specified in {sockAddr} via socket
+	// {sockfd}. Returns 0 if the send succeeds and return -1 if an error
+	// occurred.
+	//
+	// See: man 2 sendto
+	def sendto(sockfd: int, msg: Array<byte>, sockAddr: Array<byte>) -> int {
+		def ret = Linux.syscall(
+			LinuxConst.SYS_sendto, (
+				sockfd,
+				Pointer.atContents(msg),
+				msg.length,
+				0 /* flags */,
+				Pointer.atContents(sockAddr),
+				sockAddr.length
+			)
+		);
+		return if(ret.0 >= 0, int.view(ret.0), -1);
```

Comment:
Another thing to think about is how to expose finer-granularity error conditions to applications using the library. I cut a lot of corners in making `System` very very small, but a networking library should give decent errors to help troubleshoot applications.

Reply:
Right. That's fair. I was emulating the `System` component in-terms of the syscalls, but yeah there is room for providing better errors. We should add an `enum` for `errno` values and their respective error strings. It would be useful imo.